### DISCUSSION TITLE
Clarification on voltages/schematic/expansions

### DISCUSSION THREAD

**Discussion by Hyenadae** [picocomputer/community#51] [Q&A] [answered]
Hi all,
I am a fan of retro computing products such as the Ben Eater 6502 project and stumbled upon this 6502 modernization work with the Pi Pico boards. 

I have gone through a few of the videos on the project, but I am still wondering about a few things, and would be grateful to understand a bit more before ordering the pre-assembled (sockets + resistors on board) kits from Rumble/the creator, as I live overseas in Bermuda (only 800 miles away, but shipping costs/time and customs forms can be an annoyance)

Currently, I have a semi-functional 6502 Ben Eater kit with the new CMOS 6502 + 6522 VIA chip but with the orig and slow 32K ROM + 32k RAM and the LCD (got to where the ROM + 6502 was executing and writing text to the LCD, but failed to get RAM working due to wiring/programming problems sadly). From what I can tell, I can reuse my 6502 chip + 6522 since they work at 3.3v, up to 8MHz? I have a spare SN70HC30 along with other fun chips I brought before I stopped working on it, but I can then ship to my courier the few missing parts if that is the case and reuse the rest. 

I am a bit unsure about how the voltages work, I assume the Pi Pico is 3.3v on the I/O pins only, and so the whole board is 3.3v, even though USB-C can provide the 5v needed from the Vref/VBUS lines? So, I assume I would be unable to use the expansion port on the left-side (with the 20 or so pins) on 5v native devices such as the character LCD that usually directly interacted with the VIA (now at 3.3v too).

I'm mainly curious of the best way to do further expansion, or basic interfacing with other electronics, like some 5v arduino-centric sensors and audio chips such as the SN76489. I built a BBC Micro style (4mhz crystal) chiptune player with the ESP8266 + Shift register, which is a mixed 3.3v and 5v system, and there it seems you can 'cheat' 5v compatibility by also using a shift register to reduce 'CPU'-side pin usage for sending data, at the cost of code/Cycles/interrupts.

![image](https://github.com/picocomputer/discussions/assets/3067205/d371a574-607a-4007-ab58-cbab46d77831)

I have got a logic level shifter board laying around somewhere, 
![image](https://github.com/picocomputer/discussions/assets/3067205/3b14bb3d-08b8-48d1-b5c7-d9a7741c1860)
 but I assume there are some caveats to using them with this project to go from 3.3v digital logic signals to 5v?

Additionally, on the schematics I saw only a few pins from the Pi Picos are left (about 5?) plus the VIA (at 3.3v) available via J1, which is the standard and I figure good for interfacing to other chips/devices. Anyways I'd love to hear about plans for the above ideas from experts on these boards, I've got a few weeks off work in Mid December, and I'd love to have another project and reason to finish off my messily wired 3yr old 6502 kit 😄

**Comment by rumbledethumps** [comment, marked as answer]
Sounds like you have it figured out. You can use the 65C02 and 65C22 you have. They do indeed run 3.3V. That level converter should be fine.

**Reply by Hyenadae** [reply]
Okay cool, so no surprise 5V capable pins aside from the obviously marked VBUS stuff? I'll start going through my old board and chip bags before I place an order to ya 👍
